src/autoscaler/api/server/api_handler.go

```diff
@@ -131,14 +141,51 @@ func (h *ApiHandler) BindServiceInstance(w http.ResponseWriter, r *http.Request,
 		return
 	}
 
-	if instanceId == "" || bindingId == "" || body.ServiceID == "" || body.PlanID == "" {
+	if body.AppID == "" || instanceId == "" || bindingId == "" || body.ServiceID == "" || body.PlanID == "" {
 		handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
 			Code:    "Bad Request",
 			Message: "Malformed or missing mandatory data",
 		})
 		return
 	}
 
+	if body.Policy == "" {
+		h.logger.Info("no policy json provided", lager.Data{})
+	} else {
+		errResults, valid := h.policyValidator.ValidatePolicy(body.Policy)
+
+		if !valid {
+			handlers.WriteJSONResponse(w, http.StatusBadRequest, errResults)
+			return
+		}
+
+		policyGuid, err := uuid.NewV4()
+		if err != nil {
+			handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+				Code:    "Interal-Server-Error",
+				Message: "Error generating policy guid"})
+			return
+		}
+
+		h.logger.Info("saving policy json", lager.Data{"policy": body.Policy})
+		err = h.policydb.SaveAppPolicy(body.AppID, body.Policy, policyGuid.String())
+		if err != nil {
+			handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+				Code:    "Interal-Server-Error",
+				Message: "Error saving policy"})
+			return
+		}
+
+		h.logger.Info("creating/updating schedules", lager.Data{"policy": body.Policy})
+		err = h.schedulerUtil.CreateOrUpdateSchedule(body.AppID, body.Policy, policyGuid.String())
+		if err != nil {
```

Comment:
No, we still need to CreateOrUpdateSchedule before save the policy, since scheduler may return 400 bad request to reject the policy, see 
https://github.com/cloudfoundry-incubator/app-autoscaler/blob/dd581153997db77d3d82709e918a074f91aeea6f/src/autoscaler/api/schedulerutil/schedulerutil.go#L69-L72
Also,  in this situation,  we should not return InternalServerError to end-user. 
I didn't take time to check scheduler validation further. If the `400 bad request` response won't happen at all, let us delete the validation in scheduler to avoid confusion.

Reply:
I've removed the schedules validation from scheduler.